extras/beast/unit_test/dstream.hpp

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BEAST_UNIT_TEST_DSTREAM_HPP
+#define BEAST_UNIT_TEST_DSTREAM_HPP
+
+#include <boost/utility/base_from_member.hpp>
+#include <iostream>
+#include <memory>
+#include <ostream>
+#include <sstream>
+#include <string>
+
+#ifdef _MSC_VER
+# ifndef NOMINMAX
+#  define NOMINMAX 1
+# endif
+# ifndef WIN32_LEAN_AND_MEAN
+#  define WIN32_LEAN_AND_MEAN
+# endif
+# include <windows.h>
+# undef WIN32_LEAN_AND_MEAN
+# undef NOMINMAX
+#endif
+
+namespace beast {
+namespace unit_test {
+
+namespace detail {
+
+#ifdef _MSC_VER
+
+template<class CharT, class Traits, class Allocator>
+class dstream_buf
+    : public std::basic_stringbuf<CharT, Traits, Allocator>
+{
+    bool dbg_;
+
+    template<class T>
+    void write(T const*) = delete;
```

Comment:
I'm not sure what this is doing. Preventing automatic conversions to `char const*` maybe?